cradle-cassandra/src/test/java/com/exactpro/cradle/cassandra/integration/messages/GetGroupedMessageBatchesApiTest.java

```diff
@@ -157,4 +161,221 @@ public void getGroupedMessagesWithNarrowIntervalTest() throws CradleStorageExcep
+
+
+        // Expected batch2 from BaseMessageApiTest
+        var actual = storage.getGroupedMessageBatches(filter);
+        var resultAsList = Lists.newArrayList(actual.asIterable());
+        Assertions.assertThat(resultAsList.size()).isEqualTo(0);
+    }
+
+    @Test(description = "Get grouped messages with limit 1 with time borders corresponding to message in the middle of batch 8.")
+    public void getGroupedMessageWithBothTimeLimitsResultInTheMiddleOfBatch() throws CradleStorageException, IOException {
+        GroupedMessageFilter filter = GroupedMessageFilter.builder()
+                .groupName(GROUP3_NAME)
+                .bookId(bookId)
+                .timestampFrom().isGreaterThanOrEqualTo(dataStart.plus(57, ChronoUnit.MINUTES))
+                .timestampTo().isLessThan(dataStart.plus(59, ChronoUnit.MINUTES))
+                .limit(1)
+                .build();
+
+
+        // Expected batch8 from BaseMessageApiTest
+        var actual = storage.getGroupedMessageBatches(filter);
+        var resultAsList = Lists.newArrayList(actual.asIterable());
+        Assertions.assertThat(resultAsList.size()).isEqualTo(1);
+        Assertions.assertThat(resultAsList.get(0).getMessages().size()).isEqualTo(3);
+
+        Assertions.assertThat(resultAsList.get(0).getFirstTimestamp()).isEqualTo(dataStart.plus(56, ChronoUnit.MINUTES));
+        Assertions.assertThat(resultAsList.get(0).getLastTimestamp()).isEqualTo(dataStart.plus(59, ChronoUnit.MINUTES));
+    }
+
+    @Test(description = "Get grouped messages with limit 1 with time borders corresponding batch8 but not equal to first and last timestamp")
+    public void getGroupedMessageWithBothTimeLimitsCoveringWholeBatch() throws CradleStorageException, IOException {
+        GroupedMessageFilter filter = GroupedMessageFilter.builder()
+                .groupName(GROUP3_NAME)
+                .bookId(bookId)
+                .timestampFrom().isGreaterThanOrEqualTo(dataStart.plus(55 * 60 + 1, ChronoUnit.SECONDS))
+                .timestampTo().isLessThan(dataStart.plus(61, ChronoUnit.MINUTES))
+                .limit(1)
+                .build();
+
+
+        // Expected batch8 from BaseMessageApiTest
+        var actual = storage.getGroupedMessageBatches(filter);
+        var resultAsList = Lists.newArrayList(actual.asIterable());
+        Assertions.assertThat(resultAsList.size()).isEqualTo(1);
+        Assertions.assertThat(resultAsList.get(0).getMessages().size()).isEqualTo(3);
+
+        Assertions.assertThat(resultAsList.get(0).getFirstTimestamp()).isEqualTo(dataStart.plus(56, ChronoUnit.MINUTES));
+        Assertions.assertThat(resultAsList.get(0).getLastTimestamp()).isEqualTo(dataStart.plus(59, ChronoUnit.MINUTES));
+    }
+
+    @Test(description = "Get grouped messages with limit 1 with start time limit corresponding to batch5 but less than its start time.")
+    public void getGroupedMessageWithLeftLimitBeforeBatchStart() throws CradleStorageException, IOException {
```

Comment:
Looks like this test checks the same case as getGroupedMessageWithLeftLimitBeforeLastBatchStart

Reply:
getGroupedMessageWithLeftLimitBeforeLastBatchStart targets last available batch in storage, this one isn't